runner/run_solver.py

```
@@ -95,23 +96,41 @@ def get_milp_metrics(input_file, solver_result):
     duality gap.
     """
-    if highspy is not None:
-        h = highspy.Highs()
-        h.readModel(input_file)
-        integer_vars = {
-            h.variableName(i)
-            for i in range(h.numVariables)
-            if h.getColIntegrality(i)[1] == highspy.HighsVarType.kInteger
-        }
-        if integer_vars:
-            duality_gap = get_duality_gap(solver_result.solver_model, solver_name)
-            max_integrality_violation = calculate_integrality_violation(
-                integer_vars, solver_result.solution.primal
-            )
-            return duality_gap, max_integrality_violation
+    try:
+        if highspy is not None:
+            h = highspy.Highs()
+            h.readModel(input_file)
+            integer_vars = {
+                h.variableName(i)
+                for i in range(h.numVariables)
+                if h.getColIntegrality(i)[1] == highspy.HighsVarType.kInteger
+            }
+            if integer_vars:
+                duality_gap = get_duality_gap(solver_result.solver_model, solver_name)
+                max_integrality_violation = calculate_integrality_violation(
+                    integer_vars, solver_result.solution.primal
+                )
+                return duality_gap, max_integrality_violation
+    except Exception:
+        print(f"ERROR obtaining reported runtime: {format_exc()}", file=sys.stderr)
     return None, None
 
 
+def get_reported_runtime(solver_name, solver_model) -> float | None:
+    """Get the solving runtime as reported by the solver from the solver's Python object."""
+    try:
+        match solver_name:
+            case "highs":
+                return solver_model.getRunTime()
+            case "scip":
+                return solver_model.getSolvingTime()
+            case "gurobi":
+                return solver_model.Runtime
+    except Exception:
+        print(f"ERROR obtaining reported runtime: {format_exc()}", file=sys.stderr)
+    return None
+
```

Should we add a default case to raise an error for unsupported solvers?

---

Great catch, thanks! I've fixed it